### ISSUE TITLE
New bare workflow - API calls hanging

### ISSUE THREAD

**Issue by hardcodet** [expo/expo#11128]
## 🐛 Bug Report

After I ejected my app, I couldn't access any APIs anymore, and after about a day of sweating and swearing, I triangulated this down to a clean new bare workflow, so this doesn't work right from the start.

My repro is as minimal as it gets:

```
import React from 'react';
import {Button, Image, StyleSheet, Text, View} from 'react-native';

export default function App() {

  const getKitten = async () => {
    const result = await fetch("https://placekitten.com/300/300");
    alert("result: "+ result.status);
  }

  return (
    <View>
      <Image style={{width: 100, height: 100}} source={{uri: "https://placekitten.com/300/300"}}
             resizeMode={"cover"}/>
      <Button onPress={() => getKitten()} title={"get kitten"} />
    </View>
  );
}
```

That alert will never show - the `await` on the fetch promise hangs forever.
Note that the image can easily load the picture though (same URL), and shows a kitten.

I assume this is something on my machine (Win 10, Android), but then, running the exact same code on a regular bootstrapped React Native project runs without problems. So it appears that expo causes something to break.

### Environment

```
  Expo CLI 3.28.5 environment info:
    System:
      OS: Windows 10 10.0.18363
    Binaries:
      Node: 12.18.2 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
    SDKs:
      Android SDK:
        API Levels: 23, 29
        Build Tools: 28.0.3, 29.0.2
        System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6514223
    npmPackages:
      expo: ~39.0.2 => 39.0.5
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: ~0.63.3 => 0.63.3
      react-native-web: ~0.13.12 => 0.13.18
    Expo Workflow: bare
```

**Comment by brentvatne** [comment]
hi there! try adding try/catch around the `await` to see why it's failing :)

> running the exact same code on a regular bootstrapped React Native project runs without problems

the thing is, a bare project initialized with expo-cli is _a regular bootstrapped React Native project_ with only a few modules added, and none of those would have any impact on whether `fetch` would work.

<hr />

i copied your exact code into a brand new bare project initialized with expo-cli and ran it, this was the result

![Screen Shot 2020-11-25 at 1 42 32 PM](https://user-images.githubusercontent.com/90494/100284579-38034a80-2f24-11eb-92df-a2a3e0c74c08.png)

exact source is here https://github.com/brentvatne/hardcodet

**Comment by hardcodet** [comment]
@brentvatne Yes of course this will work on your machine - I don't assume expo init is broken across the board :)
I also assume it's got at least in part to do something with my setup, and I wouldn't have posted **if** a project I bootstrapped with the RN CLI would have been broken, too.

Here's what I did:

- expo managed: worked
- expo managed, then ejected: stopped working
- fresh bare project: not working
- fresh RN CLI project: working

This is all on the same machine, so the Bare libraries / app initialization flow have to play their role in this.
 There's also no exception, the call just hangs indefinitely. I'm fine not pursuing this, though.
